### ISSUE TITLE
Rename InheritanceManager2 to InheritanceHelper/Util/Something and move it out of src

### ISSUE THREAD

**Issue by kevmoo** [dart-lang/sdk#35927]
1.  `InheritanceManager2` is not a name we want to keep around. Once `InheritanceManager` is removed it'll be just...weird...
2. `InheritanceManager2` lives under `lib/src` and is not exported pub a "public" library.

Related to https://github.com/dart-lang/sdk/issues/35833

**Comment by kevmoo** [comment]
See https://dart-review.googlesource.com/c/sdk/+/92920

**Comment by srawlins** [comment]
This class is now InheritanceManager3 😛 

Can you say why you want this out of `lib/src`? Sorry for the extreme delay!

**Comment by kevmoo** [comment]
I use it here: https://github.com/google/json_serializable.dart/blob/190f8b1ea1fdd87943ba230f1deddcc31393a348/json_serializable/lib/src/field_helpers.dart#L8

dartdoc here: https://github.com/dart-lang/dartdoc/blob/2eee9a9231ff29913c693b7e1a0dc32b936b34b7/lib/src/model/library.dart#L14

So it's at thing that's still wanted

**Comment by srawlins** [comment]
I agree, we [use it in mockito](https://github.com/dart-lang/mockito/blob/master/lib/src/builder.dart) as well.

WDYT, @scheglov @bwilkerson? Move `InheritanceManager3` to `lib/dart/element/`, next to, e.g. `TypeSystem`? Could rename it to `InheritanceManager` as well, and keep a deprecated reference in the old location.

**Comment by scheglov** [comment]
Most of the methods that exist in `InheritanceManager3` should be just methods of `InterfaceElement`. So, just moving it and exposing the official API is wrong. We should design something better.

**Comment by srawlins** [comment]
SG
